File: P1030求先序排列.py
def dfs(in_, after):
    if len(in_) > 0:
        ch = after[len(after)-1]
        index = in_.find(ch)
        print(ch, end="")
        dfs(in_[:index], after[:index])
        dfs(in_[index+1:], after[index:index+len(in_[index+1:])])

in_ = input()
after = input()
dfs(in_, after)

File: P1030求先序排列.cpp
#include <iostream>
#include <string>
using namespace std;


void dfs(string in, string after)
{
    if (in.size() > 0)
    {
        char ch = after[after.size()-1];
        int index = in.find(ch);
        cout<<ch;
        dfs(in.substr(0, index), after.substr(0, index));
        dfs(in.substr(index+1), after.substr(index, in.size()-1-index));
    }
}



int main()
{
    string after;
    string in;
    cin>>in>>after;
    dfs(in, after);
    return 0;
}